doc/contributing.rst

```diff
@@ -500,11 +576,63 @@ General commit rules
    squash all commits using the script shown in
    https://github.com/nim-lang/Nim/pull/9356
 
-6. Do not mix pure formatting changes (e.g. whitespace changes, nimpretty) or
+5. Do not mix pure formatting changes (e.g. whitespace changes, nimpretty) or
    automated changes (e.g. nimfix) with other code changes: these should be in
    separate commits (and the merge on GitHub should not squash these into 1).
 
 
+Pull requests message
+---------------------
+
+Pull request description and title generally follow the same guidelines as
+commit messages.
+
+
+The `git`:cmd: stuff
+====================
+
+PR should contain a single commit
+---------------------------------
+
+Pull request should contain a single **passing** commit. This is necessary
+to ensure clean git history that is not cluttered by a partially working,
+failing and outright failing to compile states.
+
+To achieve this you can do either of the following:
+
+- If the change fits into a single commit you don't need to do anything
+- If you need to made some additional modifications (review requested) you
+  can extend the commit and force-push it (`git commit --amend
+  --no-edit`:cmd: and `git push --force <remote> <your branch>`:cmd:)
+- Create multiple commits and then squash them when your pull request is
```

Comment:
Can we elevate this one, it's a nicer flow and less frustrating. It's one of those things where people don't realize they can do this.

Reply:
It is more frustrating to do right now, requires digging into multistep git stuff create an intermediate branch. Extending the same commit and force-pushing it is much easier, especially for a small changes, where you just need to correct a couple of lines.